test/icc-x-api/icc-invoice-x-api.ts

```diff
@@ -0,0 +1,101 @@
+      await patientApiForHcp.newInstance(
+        hcpUser,
+        new Patient({
+          id: randomUUID(),
+          firstName: 'John',
+          lastName: 'Snow',
+          note: 'Winter is coming',
+        })
+      )
+    )
+  }
+
+  it('Test', async () => {
+    // Given
+    const username = env.dataOwnerDetails[hcp1Username].user
+    const password = env.dataOwnerDetails[hcp1Username].password
+
+    const authProvider = new BasicAuthenticationProvider(username, password)
+
+    const userApi = new IccUserXApi(env.iCureUrl, {}, authProvider, fetch)
+    const invoiceApi = new IccInvoiceApi(env.iCureUrl, {}, authProvider, fetch)
+
+    const currentUser = await userApi.getCurrentUser()
+
+  })
+
+  it('Test findBy not usingPost', async () => {
+    // Given
+    const {
+      userApi: userApiForHcp,
+      dataOwnerApi: dataOwnerApiForHcp,
+      patientApi: patientApiForHcp,
+      cryptoApi: cryptoApiForHcp,
+      dataOwnerApi: dateOwnerApiForHcp,
+      entityReferenceApi: entityReferenceApiForHcp,
+    } = await Api(env!.iCureUrl, env!.dataOwnerDetails[hcp1Username].user, env!.dataOwnerDetails[hcp1Username].password, crypto)
+    const hcpUser = await userApiForHcp.getCurrentUser()
+    await initKey(dataOwnerApiForHcp, cryptoApiForHcp, hcpUser, env!.dataOwnerDetails[hcp1Username].privateKey)
+
+    const username = env.dataOwnerDetails[hcp1Username].user
+    const password = env.dataOwnerDetails[hcp1Username].password
+
+    const authProvider = new BasicAuthenticationProvider(username, password)
+
+    const invoiceXApi = new IccInvoiceXApi(env.iCureUrl, {}, cryptoApiForHcp, entityReferenceApiForHcp, dateOwnerApiForHcp, authProvider, fetch)
+
+    const patient = (await createPatient(patientApiForHcp, hcpUser)) as Patient
+
+    await invoiceXApi.findBy(hcpUser.healthcarePartyId!, patient, false)
+  })
+
+  it('Test findBy usingPost', async () => {
+    // Given
+    const {
+      userApi: userApiForHcp,
+      dataOwnerApi: dataOwnerApiForHcp,
+      patientApi: patientApiForHcp,
+      cryptoApi: cryptoApiForHcp,
+      dataOwnerApi: dateOwnerApiForHcp,
+      entityReferenceApi: entityReferenceApiForHcp,
+    } = await Api(env!.iCureUrl, env!.dataOwnerDetails[hcp1Username].user, env!.dataOwnerDetails[hcp1Username].password, crypto)
+    const hcpUser = await userApiForHcp.getCurrentUser()
+    await initKey(dataOwnerApiForHcp, cryptoApiForHcp, hcpUser, env!.dataOwnerDetails[hcp1Username].privateKey)
+
+    const username = env.dataOwnerDetails[hcp1Username].user
+    const password = env.dataOwnerDetails[hcp1Username].password
+
+    const authProvider = new BasicAuthenticationProvider(username, password)
+
+    const invoiceXApi = new IccInvoiceXApi(env.iCureUrl, {}, cryptoApiForHcp, entityReferenceApiForHcp, dateOwnerApiForHcp, authProvider, fetch)
+
+    const patient = (await createPatient(patientApiForHcp, hcpUser)) as Patient
+
+    await invoiceXApi.findBy(hcpUser.healthcarePartyId!, patient, true)
```

Comment:
Add assertions on your test, to make sure you find the element you created. 
So create an instance linked to your patient and assert you find the correct one